src/Tvheadend.cpp

```diff
@@ -367,6 +367,39 @@ PVR_ERROR CTvheadend::GetChannels(bool radio, kodi::addon::PVRChannelsResultSet&
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR CTvheadend::GetChannelStreamProperties(const kodi::addon::PVRChannel& channel, std::vector<kodi::addon::PVRStreamProperty>& properties)
+{
+  CLockObject lock(m_mutex);
+
+  if (!Settings::GetInstance().GetStreamingHTTP())
+    return PVR_ERROR_NO_ERROR;
+
+  auto it = m_channels.find(channel.GetUniqueId());
+  if (it == m_channels.end())
+    return PVR_ERROR_FAILED;
+
+  std::string url = Settings::GetInstance().GetUseHTTPS() ? "https://" : "http://";
+  std::string usn = Settings::GetInstance().GetUsername();
+  std::string pas = Settings::GetInstance().GetPassword();
+
+  if (usn[0] && pas[0])
```

Comment:
I think empty() is the C++ way to check this

Reply:
Ooops.